src/render/decorations/CHyprGroupBarDecoration.cpp

```diff
@@ -305,3 +304,79 @@ void CHyprGroupBarDecoration::refreshGradients() {
 bool CHyprGroupBarDecoration::allowsInput() {
     return true;
 }
+
+bool CHyprGroupBarDecoration::onEndDragOnDeco(CWindow* m_pDraggedWindow, const Vector2D& pos) {
+
+    if (!(!g_pKeybindManager->m_bGroupsLocked                                                                                 // global group lock disengaged
```

Comment:
note to self: this entire if probably should be a function, it's used many times.